spark/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala

```diff
@@ -125,6 +124,39 @@ object VacuumCommand extends VacuumCommandImpl with Serializable {
     }.toDF("path")
   }
 
+  def getFilesFromReservoir(basePath: String,
+                            partitionColumns: Seq[String],
+                            reservoir: DataFrame): Dataset[SerializableFileStatus] = {
+    val expectedSchema = StructType(
+      Seq(
+        StructField("path", StringType),
+        StructField("length", LongType),
+        StructField("isDir", BooleanType),
+        StructField("modificationTime", LongType)
+      ))
+
+    if(reservoir.schema != expectedSchema) {
```

Comment:
Perhaps we could allow additional fields, i.e., just check for the fields that we need. If we want to expand the set of fields in the future then this would give users more flexibility when preparing for upgrades.